ckldb related compilation errors on netlrts-linux-clang and netlrts-darwin

```
/scratch/autobuild/netlrts-linux-x86_86-clang/charm/src/ck-core/../ck-ldb/TreeBuilder.h:357:9: error: no matching member function for call to 'getProperty'
        getProperty("strategies", std::vector<std::string>(), processgroup_config);
        ^~~~~~~~~~~
/scratch/autobuild/netlrts-linux-x86_86-clang/charm/src/ck-core/../ck-ldb/TreeBuilder.h:20:5: note: candidate template ignored: substitution failure [with T = std::vector<std::__cxx11::basic_string<char>, std::allocator<std::__cxx11::basic_string<char> > >]
  T getProperty(const char* property, const T defaultValue, const json& config) const = delete;
    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

Links to autobuild output:
http://charm.cs.illinois.edu/autobuild/old.2020_09_10__01_03/netlrts-linux-x86_64-clang.txt

http://charm.cs.illinois.edu/autobuild/old.2020_09_10__01_03/netlrts-darwin-x86_64.txt

--- a/src/ck-ldb/TreeBuilder.h
+++ b/src/ck-ldb/TreeBuilder.h
@@ -16,8 +16,24 @@ public:
                         std::vector<LevelLogic*>& comm_logic, json& config) = 0;
 
   virtual ~LBTreeBuilder() {}
-  template <typename T>
-  T getProperty(const char* property, const T defaultValue, const json& config) const = delete;
+
+  int getProperty(const char* property, const int defaultValue, const json& config) const
+  {
+    return getPropertyHelper(property, defaultValue, config, &json::is_number_integer);
+  }
+
+  bool getProperty(const char* property, const bool defaultValue,
+                   const json& config) const
+  {
+    return getPropertyHelper(property, defaultValue, config, &json::is_boolean);
+  }
+
+  std::vector<std::string> getProperty(const char* property,
+                                       const std::vector<std::string> defaultValue,
+                                       const json& config) const
+  {
+    return getPropertyHelper(property, defaultValue, config, &json::is_array);
+  }
 
 protected:
   int rootPE = 0;
@@ -55,31 +71,6 @@ private:
     return defaultValue;
   }
 };
-
-template <>
-int LBTreeBuilder::getProperty<int>(const char* property, const int defaultValue, const json& config) const
-{
-  return getPropertyHelper(property, defaultValue, config, &json::is_number_integer);
-}
-
-template <>
-uint8_t LBTreeBuilder::getProperty<uint8_t>(const char* property, const uint8_t defaultValue, const json& config) const
-{
-  return getPropertyHelper(property, defaultValue, config, &json::is_number_unsigned);
-}
-
-template <>
-bool LBTreeBuilder::getProperty<bool>(const char* property, const bool defaultValue, const json& config) const
-{
-  return getPropertyHelper(property, defaultValue, config, &json::is_boolean);
-}
-
-template <>
-std::vector<std::string> LBTreeBuilder::getProperty<std::vector<std::string>>(
-    const char* property, const std::vector<std::string> defaultValue, const json& config) const
-{
-  return getPropertyHelper(property, defaultValue, config, &json::is_array);
-}
 
 class PE_Root_Tree : public LBTreeBuilder
 {

